File: src/Java/Filtering/TransformPolyData.java
import vtk.vtkRenderWindow;
import vtk.vtkActor;
import vtk.vtkNativeLibrary;
import vtk.vtkPolyDataMapper;
import vtk.vtkNamedColors;
import vtk.vtkRenderer;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkSphereSource;
import vtk.vtkTransform;
import vtk.vtkTransformPolyDataFilter;


public class TransformPolyData 
{

  //-----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------
    
  public static void main(String args[]) 
  {

    vtkNamedColors colors = new vtkNamedColors();

    //For OriginalActor Color
    double OriginalActorColor[] = new double[4];
    
    //Renderer Background Color
    double Bgcolor[] = new double[4];
    
    //TransformedActor color
    double TransformedActorColor[] = new double[4];

    colors.GetColor("LightSeaGreen", OriginalActorColor);
    colors.GetColor("Sienna", Bgcolor);
    colors.GetColor("DeepSkyBlue", TransformedActorColor);

    //Create a Sphere
    vtkSphereSource Sphere = new vtkSphereSource();
    Sphere.Update();

    //Create a Mapper and Actor
    vtkPolyDataMapper OriginalMapper = new vtkPolyDataMapper();
    OriginalMapper.SetInputConnection(Sphere.GetOutputPort());

    vtkActor OriginalActor = new vtkActor();
    OriginalActor.SetMapper(OriginalMapper);
    OriginalActor.GetProperty().SetColor(OriginalActorColor);


    //Set up the transform filter
    vtkTransform Translation = new vtkTransform();
    Translation.Translate(1.0, 2.0, 3.0);

    vtkTransformPolyDataFilter TransformFilter = new vtkTransformPolyDataFilter();
    TransformFilter.SetInputConnection(Sphere.GetOutputPort());
    TransformFilter.SetTransform(Translation);
    TransformFilter.Update();

    //Set up the actor to display the transformed polydata
    vtkPolyDataMapper TransformedMapper = new vtkPolyDataMapper();
    TransformedMapper.SetInputConnection(TransformFilter.GetOutputPort());

    vtkActor TransformedActor = new vtkActor();
    TransformedActor.SetMapper(TransformedMapper);
    TransformedActor.GetProperty().SetColor(TransformedActorColor);

    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);

    // Visualize the Actor
    ren.AddActor(OriginalActor);
    ren.AddActor(TransformedActor);
    ren.SetBackground(Bgcolor);

    renWin.SetSize(300, 300);
    renWin.Render();

    iren.Initialize();
    iren.Start();

  }
}

File: src/Python/Filtering/TransformPolyData.py
#!/usr/bin/env python

import vtk

def main():
    colors = vtk.vtkNamedColors()
    
    # Create the polydata geometry
    sphereSource = vtk.vtkSphereSource()
    sphereSource.Update()
    
    # Set up the actor to display the untransformed polydata
    originalMapper = vtk.vtkPolyDataMapper()
    originalMapper.SetInputConnection(sphereSource.GetOutputPort())
    
    originalActor = vtk.vtkActor()
    originalActor.SetMapper(originalMapper)
    originalActor.GetProperty().SetColor(colors.GetColor3d("blue"))
    
    # Set up the transform filter
    translation = vtk.vtkTransform()
    translation.Translate(1.0, 2.0, 3.0)
    
    transformFilter = vtk.vtkTransformPolyDataFilter()
    transformFilter.SetInputConnection(sphereSource.GetOutputPort())
    transformFilter.SetTransform(translation)
    transformFilter.Update()
    
    # Set up the actor to display the transformed polydata
    transformedMapper = vtk.vtkPolyDataMapper()
    transformedMapper.SetInputConnection(transformFilter.GetOutputPort())
    
    transformedActor = vtk.vtkActor()
    transformedActor.SetMapper(transformedMapper)
    transformedActor.GetProperty().SetColor(colors.GetColor3d("red"))
    
    # Set up the rest of the visualization pipeline
    renderer = vtk.vtkRenderer()
    renderer.AddActor(originalActor)
    renderer.AddActor(transformedActor)
    renderer.SetBackground(colors.GetColor3d("green")) # Set renderer's background color to green
    
    renderWindow = vtk.vtkRenderWindow()
    renderWindow.AddRenderer(renderer)
    renderWindowInteractor = vtk.vtkRenderWindowInteractor()
    renderWindowInteractor.SetRenderWindow(renderWindow)
    renderWindowInteractor.Start()
	
if __name__ == '__main__':
    main()
